[FAIL]
Traceback (most recent call last):
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/test/test_tcp.py", line 266, in test_disconnectEvent2
    self.assertEquals([], errors)
twisted.trial.unittest.FailTest: not equal:
a = []
b = [IOError(9, 'Bad file descriptor')]


twisted.internet.test.test_tcp.TCPClientTestsBuilder_EPollReactor.test_disconnectEvent2
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/test/test_tcp.py", line 266, in test_disconnectEvent2
    self.assertEquals([], errors)
twisted.trial.unittest.FailTest: not equal:
a = []
b = [ValueError('file descriptor cannot be a negative integer (-1)',)]


twisted.internet.test.test_tcp.TCPClientTestsBuilder_Glib2Reactor.test_disconnectEvent2
twisted.internet.test.test_tcp.TCPClientTestsBuilder_Gtk2Reactor.test_disconnectEvent2
twisted.internet.test.test_tcp.TCPClientTestsBuilder_PollReactor.test_disconnectEvent2
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/base.py", line 426, in _continueFiring
    callable(*args, **kwargs)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/base.py", line 609, in disconnectAll
    selectables = self.removeAll()
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/pollreactor.py", line 142, in removeAll
    [self._selectables[fd] for fd in self._writes])
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/posixbase.py", line 491, in _removeAll
    self.removeWriter(writer)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/pollreactor.py", line 134, in removeWriter
    return self._dictRemove(writer, self._writes)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/pollreactor.py", line 106, in _dictRemove
    self._updateRegistration(fd)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/pollreactor.py", line 72, in _updateRegistration
    self._poller.unregister(fd)
exceptions.ValueError: file descriptor cannot be a negative integer (-1)

twisted.internet.test.test_tcp.TCPClientTestsBuilder_PollReactor.test_disconnectEvent2
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/python/log.py", line 84, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/python/log.py", line 69, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/python/context.py", line 59, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/python/context.py", line 37, in callWithContext
    return func(*args,**kw)
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/tcp.py", line 514, in connectionLost
    self._closeSocket()
  File "/mnt/part10/opt/python/Twisted/trunk.build/build/lib.linux-x86_64-2.6/twisted/internet/tcp.py", line 92, in _closeSocket
    skt = self.socket
exceptions.AttributeError: 'Server' object has no attribute 'socket'

twisted.internet.test.test_tcp.TCPClientTestsBuilder_SelectReactor.test_disconnectEvent2
